Makefile

```diff
@@ -4,24 +4,42 @@ LDFLAGS =
 LDLIBS = -lm
 INCLUDES =
 CFLAGS_COND = -march=native
-
-# Find nvcc
-NVCC := $(shell which nvcc 2>/dev/null)
+SHELL_UNAME = $(shell uname)
+REMOVE_FILES = rm -f
+OUTPUT_FILE = -o $@
 
 # NVCC flags
 NVCC_FLAGS = -O3 --use_fast_math
 NVCC_LDFLAGS = -lcublas -lcublasLt
 
-# Function to test if the compiler accepts a given flag.
-define check_and_add_flag
+ifneq ($(OS), Windows_NT)
+  NVCC := $(shell which nvcc 2>/dev/null)
+
+  # Function to test if the compiler accepts a given flag.
+  define check_and_add_flag
     $(eval FLAG_SUPPORTED := $(shell printf "int main() { return 0; }\n" | $(CC) $(1) -x c - -o /dev/null 2>/dev/null && echo 'yes'))
     ifeq ($(FLAG_SUPPORTED),yes)
         CFLAGS += $(1)
     endif
-endef
+  endef
 
-# Check each flag and add it if supported
-$(foreach flag,$(CFLAGS_COND),$(eval $(call check_and_add_flag,$(flag))))
+  # Check each flag and add it if supported
+  $(foreach flag,$(CFLAGS_COND),$(eval $(call check_and_add_flag,$(flag))))
+else
+  CFLAGS :=
+  REMOVE_FILES = del
+  SHELL_UNAME := Windows
+  NVCC := $(shell where nvcc 2> nul)
+  CC := cl
+  CFLAGS = /Idev /Zi /nologo /Wall /WX- /diagnostics:column /sdl /O2 /Oi /Ot /GL /D _DEBUG /D _CONSOLE /D _UNICODE /D UNICODE /Gm- /EHsc /MD /GS /Gy /fp:fast /Zc:wchar_t /Zc:forScope /Zc:inline /permissive- \
```

Comment:
Perhaps add a flag that is only active during windows build..
/D WIN_BUILD or something like that.

Reply:
There's plenty of predefined macros by the compiler to know that we're building on Windows. Use those?

Reply:
Yes, but since they are so many, and perhaps differ on different compiler versions (who knows), and since we control the build, we can easily just declare one that we know will be used in this codebase.

Reply:
Probably _WIN32 or _WIN64. I've seen _MSC_VEW used a bit too.

Reply:
I think it is better we set a standard in our own code for this in the codebase to just distinguish between Windows and non-Windows builds for clarity and control.

Reply:
I would opt for the predefined _WIN64 but don't have a strong preference.

Reply:
I think that only works if we compile for the 64-bit case right?
Maybe we don't need to distinguish between those (yet).
_WIN32 is defined for both 32-bit and 64-bit environments when compiling on Windows, so it is more universal.
So if we don't use a custom macro, then I would opt for the _WIN32. 
But I would still like to see our own custom macro, because of the reasons i mentioned.

Reply:
Got it - _WIN32 works and another TBD name-wise.